src/WebJobs.Script/Workers/ProcessManagement/WorkerProcess.cs

```diff
@@ -68,8 +70,11 @@ public Task StartProcessAsync()
                     // Register process only after it starts
                     _processRegistry?.Register(_process);
 
-                    _processMonitor = new ProcessMonitor(_process, SystemEnvironment.Instance);
-                    _processMonitor.Start();
+                    var processMonitor = _serviceProvider.GetHostServiceOrNull<IHostProcessMonitor>();
```

Comment:
Because all the OOP types are WebHost level services and the DC types are ScriptHost level services, we must use IServiceProvider to access services for the currently running host.

Reply:
I remember I was having DI issues on startup when I had tried taking IHostProcessMonitor via DI. I believe the issue is that IRpcWorkerProcessFactory is added in AddCommonServices and is scoped beyond a single host instance, whereas IHostProcessMonitor is added by the ConfigureWebJobsCall so is a ScriptHost level service. So attempting to have the factory depend on a service that is only added at the ScriptHost level won't work. Hence my need for deferred resolution.

I've verified with this code that things work as expected, in restart/error scenarios. When a new ScriptHost is started, existing OOP worker processes are disposed and this registration code runs anew, registering the new child process with the current (new) host process monitor.